### PR TITLE
feat(aws_instance): treat missing data on cloudwatchmetric

### PR DESCRIPTION
**Pull request by haidargit**
## what

The PR enables the [treat_missing_data](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/cloudwatch_metric_alarm#treat_missing_data) argument for the [cloudwatch metric alarm resource](https://github.com/cloudposse/terraform-aws-ec2-instance/blob/main/cloudwatch-alarm.tf#L11) on this ec2 module.
https://github.com/cloudposse/terraform-aws-ec2-instance/blob/bf54345e18c6c2d1d85997d3af3b73a0e03c489f/cloudwatch-alarm.tf#L11

## why
Users may have specific run schedules, triggering alerts in CloudWatch, e.g., for missing data during offline periods (instance stopped state, hibernate, restart, etc.). 
This `treat_missing_data` variable will enable the CloudWatch alarms to set custom responses, which are "`missing`", "`ignore`", "`breaching`" and "`notBreaching`", for the data gaps that occur and keep the EC2 alarms condition in a desired state.

## references

- closes feature request #130 

<br>Kindly review this PR for any possible improvements. Thank you

### CODE REVIEW

**Comment by haidargit** [comment]
Hi @johncblandii, @joe-niland
could you help review the PR for the improvement? Thanks!

**Review by joe-niland — COMMENTED** [review]
@haidargit looks good. I think it would help to add [input validation](https://developer.hashicorp.com/terraform/language/expressions/custom-conditions#input-variable-validation) for this variable.

File: variables.tf [hunk 1]
```diff
@@ -283,6 +283,10 @@ variable "metric_treat_missing_data" {
   type        = string
   description = "Sets how this alarm is to handle missing data points. The following values are supported: `missing`, `ignore`, `breaching` and `notBreaching`. Defaults to `missing`."
   default     = "missing"
+  validation {
+    condition     = var.metric_treat_missing_data == "missing" && var.metric_treat_missing_data == "ignore" && var.metric_treat_missing_data == "breaching" && var.metric_treat_missing_data == "notBreaching"
+    error_message = "Only following values are supported: `missing`, `ignore`, `breaching`, and `notBreaching`. By default, it is set to `missing`."
+  }
```
Comment (joe-niland) [inline variables.tf]:
```suggestion
  validation {
    condition     = contains(["missing", "ignore", "breaching", "notBreaching"], var.metric_treat_missing_data)
    error_message = "The value of metric_treat_missing_data must be one of the following: \"missing\", \"ignore\", \"breaching\", and \"notBreaching\"."
  }
```

**Comment by haidargit** [comment]
Validation added, @joe-niland, Thanks for the correction!

**Comment by haidargit** [comment]
[variable](https://github.com/cloudposse/terraform-aws-ec2-instance/pull/184/files#diff-07f337e38ed1996f30ee4f04357ab23318c254cb606aaf4745d564cbd546722dR88) added on `examples/complete/main.tf`

Thanks

**Review by joe-niland — APPROVED** [review]

**Pull request merged**